### PR TITLE
Add save entry (collection) as datapackage

### PR DESCRIPTION
**Pull request by DunklesArchipel**
Checklist
* [x] Added an entry in `doc/news/`. 
* [x] Added a test for this change.
* [ ] Adapted the documentation for this change.

### CODE REVIEW

File: unitpackage/collection.py [hunk 1]
```diff
@@ -251,3 +251,20 @@ def __getitem__(self, identifier):
...
+        Saves all entries from a Collection as datapackages (CSV and JSON)
+        to a specific output directory :param outdir:.
+
+        EXAMPLES::
+
+            >>> db = Collection.create_example()
+            >>> db.save_entries(outdir='test/generated/saved_collection/')
+            >>> import glob
+            >>> glob.glob('test/generated/saved_collection/**.json') # doctest: +NORMALIZE_WHITESPACE
+            ['test...
```
Comment (DunklesArchipel) [inline unitpackage/collection.py:266]: For all the tests where files are generated, we should adopt an approach used in `svgdigitizer`, where we compare the content of the output files to that of the expected files.

File: unitpackage/collection.py [hunk 2]
```diff
@@ -251,3 +251,20 @@ def __getitem__(self, identifier):
                 f"The collection has more than one entry with identifier '{identifier}'."
             )
         return entries[0]
+
+    def save_entries(self, outdir=None):
```
Comment (DunklesArchipel) [inline unitpackage/collection.py:255]: We might want to add possibilities to save selected entries in specific subfolders.

We could define an `outdir_pattern` which creates an outdir from descriptors within an entry.

```python
def pattern(outdir, entry):
    return os.path.join(outdir, entry.user) # stores entries based on the user who created the package

def save_entries(self, outdir=None, outdir_pattern=pattern):
    for entry in self:
        if outdir_pattern:
            outdir = outdir_pattern(outdir, entry)
            entry.save(outdir=outdir)
``` 

We could implement a similar approach for the filename.

File: unitpackage/entry.py [hunk 3]
```diff
@@ -510,3 +510,57 @@ def plot(self, x_label=None, y_label=None, name=None):
...
+            >>> entry = Entry.create_examples()[0]
+            >>> entry.save(outdir='test/generated/')
+            >>> filename = entry.identifier
+            >>> os.path.exists(f'test/generated/{filename}.json') and os.path.exists(f'test/generated/{filename}.csv')
+            True
+
+        Specify a new filename without file extension.
+        According to the frictionless specification,
+        'this MUST be lower-case and contain only alphanumeric
+        characters along with ".", "_" or "-" characters'::
```
Comment (saraedum) [inline unitpackage/entry.py:532]:
```suggestion
        When a ``basename`` is set, the files are named ``basename.csv`` and ``basename.json``.
        Note that for a valid frictionless package this base name
        'MUST be lower-case and contain only alphanumeric
        characters along with ".", "_" or "-" characters'::
```

File: unitpackage/entry.py [hunk 4]
```diff
@@ -510,3 +510,57 @@ def plot(self, x_label=None, y_label=None, name=None):
...
+            >>> entry = Entry.create_examples()[0]
+            >>> filename = 'save_filename'
+            >>> entry.save(filename=filename, outdir='test/generated/')
+            >>> os.path.exists(f'test/generated/{filename}.json') and os.path.exists(f'test/generated/{filename}.csv')
+            True
+
+        """
+        if not outdir:
+            raise ValueError("An output directory `outdir` must be specified.")
+
```
Comment (saraedum) [inline unitpackage/entry.py:544]:
```suggestion
```

Fixed by the proposed interface.

**Review by saraedum — APPROVED** [review]

File: unitpackage/entry.py [hunk 5]
```diff
@@ -510,3 +510,57 @@ def plot(self, x_label=None, y_label=None, name=None):
         fig.update_yaxes(showline=True, mirror=True)
 
         return fig
+
+    def save(self, filename=None, outdir=None):
```
[the remark this answers is not in the document]
Comment (DunklesArchipel) [reply unitpackage/entry.py:514]: Cool thing with the *. I did not know that.

**Pull request merged**